packages/salesforcedx-vscode-core/src/commands/stopApexDebugLogging.ts

```diff
@@ -1,90 +1,63 @@
-/*
- * Copyright (c) 2017, salesforce.com, inc.
- * All rights reserved.
- * Licensed under the BSD 3-Clause license.
- * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
- */
-import { CommandOutput, Command, SfCommandBuilder } from '@salesforce/salesforcedx-utils';
 import {
-  CliCommandExecutor,
-  workspaceUtils,
-  CancelResponse,
-  ContinueResponse,
-  ParametersGatherer
+  ChannelService,
+  TraceFlagsRemover
 } from '@salesforce/salesforcedx-utils-vscode';
 import * as vscode from 'vscode';
-import { hideTraceFlagExpiration } from '../decorators';
-import { nls } from '../messages';
+import { OUTPUT_CHANNEL } from '../channels';
+import { WorkspaceContext } from '../context';
 import { telemetryService } from '../telemetry';
 import { developerLogTraceFlag } from '.';
-import { SfCommandlet, SfCommandletExecutor, SfWorkspaceChecker } from './util';
+import { handleStartCommand, handleFinishCommand } from '../utils/channelUtils';
 
-export class StopApexDebugLoggingExecutor extends SfCommandletExecutor<{}> {
-  public build(): Command {
-    return deleteTraceFlag();
-  }
-
-  public execute(response: ContinueResponse<{}>): void {
-    const startTime = process.hrtime();
-    const cancellationTokenSource = new vscode.CancellationTokenSource();
-    const cancellationToken = cancellationTokenSource.token;
-
-    const execution = new CliCommandExecutor(this.build(), {
-      cwd: workspaceUtils.getRootWorkspacePath()
-    }).execute(cancellationToken);
-
-    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
-    execution.processExitSubject.subscribe(async data => {
-      this.logMetric(execution.command.logName, startTime);
-      if (data !== undefined && String(data) === '0') {
-        developerLogTraceFlag.turnOffLogging();
-        hideTraceFlagExpiration();
-      }
-    });
-  }
-}
+const command = 'SFDX: Turn Off Apex Debug Log for Replay Debugger';
 
 export const turnOffLogging = async (): Promise<void> => {
+  console.log('Enter turnOffLogging()');
+  const channelService = new ChannelService(OUTPUT_CHANNEL);
+  handleStartCommand(channelService, command);
+
   if (developerLogTraceFlag.isActive()) {
-    const execution = new CliCommandExecutor(deleteTraceFlag(), {
-      cwd: workspaceUtils.getRootWorkspacePath()
-    }).execute();
-    telemetryService.sendCommandEvent(execution.command.logName);
-    const resultPromise = new CommandOutput().getCmdResult(execution);
-    const result = await resultPromise;
-    const resultJson = JSON.parse(result);
-    if (resultJson.status === 0) {
-      return Promise.resolve();
-    } else {
-      return Promise.reject('Restoring the debug levels failed.');
-    }
-  }
-};
+    console.log('Developer log trace flag is active');
+    try {
+      await vscode.window.withProgress(
```

Comment:
I don't know the UI patterns well enough to know the right choice...but does this still take long enough that you'd want a progress indicator, especially when it's not going to do anything?